components/sync15/src/request.rs

```diff
@@ -244,24 +251,20 @@ where
             last_modified: ts,
             post_limits: LimitTracker::new(config.max_post_bytes, config.max_post_records),
             batch_limits: LimitTracker::new(config.max_total_bytes, config.max_total_records),
-            batch: BatchState::NoBatch,
+            batch: BatchState::NewBatch,
             max_payload_bytes: config.max_record_payload_bytes,
             max_request_bytes: config.max_request_bytes,
             queued: Vec::new(),
         }
     }
 
-    #[inline]
-    fn in_batch(&self) -> bool {
-        match &self.batch {
-            BatchState::Unsupported | BatchState::NoBatch => false,
-            _ => true,
-        }
-    }
 
+    /// Returns Ok(true) iff a record is added with no error
```

Comment:
Describe what the function does. Also say under what circumstances it will return false (if the record can't be added given the current server limits). Also, it can return an error (from line 334, for example).

Reply:
Good, please add a brief sentence about when it can throw an error. :)